request_test.go

```diff
@@ -2861,4 +2866,39 @@ func TestRequestRetry(t *testing.T) {
 			assert.Equal(t, int64(100+200+300), totalSleepTime.Milliseconds())
 		})
 	})
+
+	t.Run("cancelled retries", func(t *testing.T) {
```

Comment:
Ideally we should also check that sleep function was NOT called.

Also would be nice to add a test when we call cancel after first retry, before next try.

Reply:
> Ideally we should also check that sleep function was NOT called.

added

btw i can't mock the sleepFn then sum the duration, because (it looks like) the sleepFn is still called in the select statement (on this line [request.go](https://github.com/gavv/httpexpect/pull/198/files#diff-3c2feffcf57ffb6c596dc5608186c518252092b7017be5189d3dc12aa3c45cecR1931)). So, i measure the elapsed time instead.

> Also would be nice to add a test when we call cancel after first retry, before next try.

this is already covered in the e2e test. would that be enough? thanks

Reply:
Oh, I see.

Yes, e2e tests are good, let's keep this unit test as is.

I think this check:

```
assert.Equal(t, time.Duration(0), elapsed.Round(req.minRetryDelay))
```

is not really useful, and also I'd prefer to avoid relying on private fields in tests when possible. So I suggest to remove it.

Also I think we should set retry delay to some large value to ensure that it wont happen before cancellation.

Reply:
got it